lib/matplotlib/axes/_axes.py

```diff
@@ -2488,6 +2491,12 @@ def bar(self, x, height, width=0.8, bottom=None, *, align="center",
                 itertools.cycle(mcolors.to_rgba_array(edgecolor)),
                 # Fallback if edgecolor == "none".
                 itertools.repeat('none'))
+        if hatchcolor is None:
+            hatchcolor = itertools.repeat(None)
```

Comment:
This appears to advance the default color so by default the hatch is different from the edge color?  That doesn't seem a reasonable default.